trin-history/src/jsonrpc.rs

```diff
@@ -107,21 +107,40 @@ async fn recursive_find_content(
             None
         }
     };
-    let (possible_content_bytes, utp_transfer, trace) = match local_content {
+    let (content_bytes, utp_transfer, trace) = match local_content {
         Some(val) => {
             let local_enr = overlay.local_enr();
             let mut trace = QueryTrace::new(&overlay.local_enr(), content_key.content_id());
             trace.node_responded_with_content(&local_enr);
-            (Some(val), false, if is_trace { Some(trace) } else { None })
+            (val, false, if is_trace { Some(trace) } else { None })
         }
-        None => overlay.lookup_content(content_key.clone(), is_trace).await,
+        // data is not available locally, make network request
+        None => match overlay
+            .lookup_content(content_key.clone(), is_trace)
+            .await
+            .expect("XXXX fix this")
+        {
+            Ok((content_bytes, utp_transfer, trace)) => (content_bytes, utp_transfer, trace),
+            Err(err) => match err.clone() {
+                OverlayRequestError::ContentNotFound((_msg, _utp, _trace)) => {
+                    // this should be formatted...
+                    return Err(err.to_string());
```

Comment:
@ogenev I'm not sure how to handle this. If content is not found on the network, the overlay now returns this `OverlayRequestError::ContentNotFound()`. However, this method (`recursive_find_content()`) returns `Result<Value, String>`. But, inside `rpc/src/fetch.rs` we want to distinguish between `ContentNotFound` errors & regular errors (eg that only contain an error message), so that we can return the appropriate `RpcServeError` with its own code / message / data fields in the case that data is not available on the network.

It seems to me like the only option here is to serialize `OverlayRequestError::ContentNotFound` to a string here, and then **try** to deserialize the string to `RpcServeError::ContentNotFound` inside `rpc/src/fetch.rs` & if that fails, then handle it as a regular `RpcServeError::Message`? 

Any thoughts? Or suggestions on how to better approach this?

Reply:
Yeah, I think this would be the simplest approach - returning `String` here and then converting it to the appropriate error message in `fetch.rs`.